### PR TITLE
v1.8.0 CHANGELOG and UpgradeGuide 

### PR DESCRIPTION
**Pull request by Zuhairahmed**
## Description

Upgrade Guide for v1.8.0

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code

### CODE REVIEW

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+sidebar_current: "docs-mongodbatlas-guides-180-upgrade-guide"
+description: |-
+MongoDB Atlas Provider 1.8.0: Upgrade and Information Guide
+---
+
+# MongoDB Atlas Provider 1.8.0: Upgrade and Information Guide
+
+The Terraform MongoDB Atlas Provider version 1.8.0 has a number of new and exciting features and changes.
+
+**Major Enhancements:**
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]: Curious why change this from simply "New Features" as it's been for quite a few versions?

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+* The HashiCorp Terraform MongoDB Atlas Provider now supports [Snapshot Distribution](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_backup_schedule) 
+  * To learn more about this feature see here: https://www.mongodb.com/blog/post/introducing-snapshot-distribution-atlas
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]: We normally start these with a "You can now..." as that was feedback from docs when they were able to review.  It makes it a bit more personal.  Also you can work in the docs link for the feature into the context to make it a bit less cluttered.  Providing example to help illustrate
```suggestion
* You can now enable and manage backup [snapshot distribution](https://www.mongodb.com/blog/post/introducing-snapshot-distribution-atlas) with [`mongodbatlas_cloud_backup_schedule`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_backup_schedule)
```

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+# MongoDB Atlas Provider 1.8.0: Upgrade and Information Guide
+
+The Terraform MongoDB Atlas Provider version 1.8.0 has a number of new and exciting features and changes.
+
+**Major Enhancements:**
+* The HashiCorp Terraform MongoDB Atlas Provider now supports [Snapshot Distribution](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_backup_schedule) 
+  * To learn more about this feature see here: https://www.mongodb.com/blog/post/introducing-snapshot-distribution-atlas   
+* The HashiCorp Terraform MongoDB Atlas Provider now allows users to [Programmatically Create API Keys](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/api_key) 
+  * To learn more about this feature see here: https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Programmatic-API-Keys
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]: Similar to the above:
```suggestion
* You can now manage the lifecycle of your [Programmatic API Keys](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Programmatic-API-Keys) [`mongodbatlas_api_key`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/api_key) 
```

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  *  **IMPORTANT WARNING:**  Creating, Reading, Updating, or Deleting Atlas API Keys may expose sensitive organizational secrets to Terraform State. Consider storing sensitive API Key secrets instead via the [HashiCorp Vault MongoDB Atlas Secrets Engine](https://developer.hashicorp.com/vault/docs/secrets/mongodbatlas).
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]: Providing a bit more here will be helpful to users.  Basically first point them to Terraform best practices and then give them another option.  
```suggestion
  *  **IMPORTANT WARNING:**  Managing Atlas Programmatic API Keys (PAKs) with Terraform will expose sensitive organizational secrets in Terraform's state.  We suggest following [Terraform's best practices](https://developer.hashicorp.com/terraform/language/state/sensitive-data).  You may also want to consider managing your PAKs via a more secure method, such as the [HashiCorp Vault MongoDB Atlas Secrets Engine](https://developer.hashicorp.com/vault/docs/secrets/mongodbatlas). 
```

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  * To learn more about this feature see here: https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Programmatic-API-Keys
+  *  **IMPORTANT WARNING:**  Creating, Reading, Updating, or Deleting Atlas API Keys may expose sensitive organizational secrets to Terraform State. Consider storing sensitive API Key secrets instead via the [HashiCorp Vault MongoDB Atlas Secrets Engine](https://developer.hashicorp.com/vault/docs/secrets/mongodbatlas). 
+
+
+**Depreciations:**
+* [`cloud_provider_snapshot`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot), [`cloud_provider_snapshot_backup_policy`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot_backup_policy), [`cloud_provider_snapshot_restore_job`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot_restore_job), and [`private_ip_mode`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/private_ip_mode) are now deprecated and will be removed from codebase as of v1.9
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]: I would make these two bullets.  I would also let the user know cloud_backup.... is the replacement for each of these, similar to the message we have in the docs for these.  I would also change the lang to make clear we deprecated these a while ago.   Make sure if you use the suggestion to replace >>> and <<<
```suggestion
* [`mongodbatlas_cloud_provider_snapshot`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot), [`mongodbatlas_cloud_provider_snapshot_backup_policy`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot_backup_policy), [`mongodbatlas_cloud_provider_snapshot_restore_job`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cloud_provider_snapshot_restore_job) have been deprecated for a few versions.  We will proceed with removal in the next major version, 1.9.  They have been replaced by >>>add the 3 new ones in the same order as above<<<< respectively. 
* [`mongodbatlas_private_ip_mode`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/private_ip_mode) has been deprecated for a few versions.  We will proceed with removal in the next major version, 1.9.  
```

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* NEW_RELIC and FLOWDOCK in [`third_party_integration`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/third_party_integration#argument-reference) resource are now deprecated and will be removed from codebase as of v1.9 release
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]: Please ensure the docs page for the resource and data sources mark these as deprecated: https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/third_party_integration.  Assuming we also marked these as deprecated in code?  

Removed codebase (and from above) as that's not how users refer to it but how we, when developing, do. 
```suggestion
* NEW_RELIC and FLOWDOCK in [`mongodbatlas_third_party_integration`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/third_party_integration#argument-reference) resource are now deprecated and will be removed in the next major version, 1.9.
```

**Review by themantissa — CHANGES_REQUESTED** [review]
Let me know if you have any questions.  Also dont' forget to add the smaller adds and I'll review those once added.

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 1 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: reverted back to "New Features" to keep consistent with prior releases. wanted to highlight that we are summarizing a subset of new features, but agree rather not confuse current users with new terminology

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 2 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 3 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 4 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+sidebar_current: "docs-mongodbatlas-guides-180-upgrade-guide"
+description: |-
+MongoDB Atlas Provider 1.8.0: Upgrade and Information Guide
+---
+
+# MongoDB Atlas Provider 1.8.0: Upgrade and Information Guide
+
+The Terraform MongoDB Atlas Provider version 1.8.0 has a number of new and exciting features and changes.
+
+**Major Enhancements:**
```
Comment (Zuhairahmed) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]:
```suggestion
**New Features:**
```

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 6 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated. pulled in additional commits to reflect changes to third_party_integrations doc pages as well. 

also see here for deprecations in each of the respective code bases: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/988/files. 

also see here for details for ticket confirming underlying resource third party integrations deprecations: https://jira.mongodb.org/browse/HELP-39527

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 5 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 4 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: also updated associated warning in terraform docs page

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  *  **IMPORTANT WARNING:**  Managing Atlas Programmatic API Keys (PAKs) with Terraform will expose sensitive organizational secrets in Terraform's state.  We suggest following [Terraform's best practices](https://developer.hashicorp.com/terraform/language/state/sensitive-data).  You may also want to consider managing your PAKs via a more secure method, such as the [HashiCorp Vault MongoDB Atlas Secrets Engine](https://developer.hashicorp.com/vault/docs/secrets/mongodbatlas). 
+* You can now retrieve `org_id` from API Keys with [`mongodbatlas_roles_org_id`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/data-sources/roles_org_id)
+* You can now use the `oplogMinRetentionHours` paramter in [`mongodbatlas_advanced_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) and [`mongodbatlas_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster) resources
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]:
```suggestion
* You can now set the [`oplogMinRetentionHours`](https://www.mongodb.com/docs/upcoming/reference/configuration-options/#mongodb-setting-storage.oplogMinRetentionHours) parameter in [`mongodbatlas_advanced_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/advanced_cluster) and [`mongodbatlas_cluster`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/cluster) resources  
```

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+* [`mongodbatlas_private_ip_mode`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/private_ip_mode) has been deprecated for a few versions.  We will proceed with removal in the next major version, 1.9.0.  
+* NEW_RELIC and FLOWDOCK in [`mongodbatlas_third_party_integration`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/third_party_integration#argument-reference) resource are now deprecated and will be removed in the next major version, 1.9.0.
```
Comment (themantissa) [inline website/docs/guides/1.8.0-upgrade-guide.html.markdown]:
```suggestion
* NEW_RELIC and FLOWDOCK in the [`mongodbatlas_third_party_integration`](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/third_party_integration#argument-reference) resource are now deprecated and will be removed in the next major version, 1.9.0.
```

**Review by themantissa — CHANGES_REQUESTED** [review]
Just a couple of tiny adjustments and it will be ready to go!

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 8 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated, makes sense including link to parameter in our docs. very helpful

File: website/docs/guides/1.8.0-upgrade-guide.html.markdown (hunk 9 above)
Comment (Zuhairahmed) [reply website/docs/guides/1.8.0-upgrade-guide.html.markdown]: updated

**Comment by Zuhairahmed** [comment]
> Just a couple of tiny adjustments and it will be ready to go!

Thanks @themantissa! Updated

**Review by themantissa — APPROVED** [review]
LGTM! Thank you

**Review by martinstibbe — APPROVED** [review]
LGTM

**Pull request merged**